lib/streamaggr/streamaggr.go

```diff
@@ -840,13 +847,18 @@ func (a *aggregator) Push(tss []prompbmarshal.TimeSeries, matchIdxs []byte) {
 		key := bytesutil.InternBytes(buf)
 		for _, sample := range ts.Samples {
 			if math.IsNaN(sample.Value) {
+				a.ignoredNanSamples.Inc()
 				// Skip NaN values
 				continue
 			}
 			if ignoreOldSamples && sample.Timestamp < minTimestamp {
+				a.ignoredOldSamples.Inc()
 				// Skip old samples outside the current aggregation interval
 				continue
 			}
+			if sample.Timestamp > 0 {
```

Comment:
@AndrewChubatiuk this looks like expensive operation in the hot path, because all samples are expected to have Timestamp.
Shall we update the histogram once per batch with the biggest lag?